lib/github-package.js

```diff
@@ -415,9 +419,12 @@ export default class GithubPackage {
   @autobind
   async cloneRepositoryForProjectPath(remoteUrl, projectPath) {
     const context = this.contextPool.getContext(projectPath);
+    // TODO: do we need to pass pipelineManager? or is this discarded immediately & recreated later in workdir context
     const repository = context.isPresent() ? context.getRepository() : new Repository(projectPath);
 
     await repository.clone(remoteUrl);
+    // TODO: ask Ash
+    // repository.destroy()
```

Comment:
Hah, this is tricky because we only _need_ to call `.destroy()` here if the `: new Repository(projectPath)` branch is followed up above. Otherwise, we already had a `WorkdirContext` in the pool for the path we're trying to clone to, and it'll be destroyed by the pool when `getNextContext()` fires. We _can't_ destroy it if the `context.getRepository()` branch is followed and that Repository is in a state like Present, because we'd still need it for further operations in that git repo... but, if it's Present, the `.clone()` call should throw anyway. So I think this line is safe 🤞